lightly/api/api_workflow_client.py

```diff
@@ -137,7 +137,7 @@ def dataset_id(self, dataset_id: str):
         Raises:
             ValueError if the dataset id does not exist.
         """
-        all_datasets: List[DatasetData] = self._datasets_api.get_datasets()
+        all_datasets: List[DatasetData] = self.get_all_datasets()
```

Comment:
Is it actually really needed to get ALL and then see if one has it? Wouldn't it be better to just try and getDatasetById and if its there, its there otherwise fail?